sdk/scheduler/src/test/java/com/mesosphere/sdk/http/EndpointUtilsTest.java

```diff
@@ -33,49 +35,73 @@ public void testToEndpoint() {
-        assertEquals("task.pathtosvc.what.a.fun.test.tld:5", EndpointUtils.toAutoIpEndpoint("/path/to/svc", "task", 5, mockSchedulerConfig));
-        assertEquals("task.pathtosvc.what.a.fun.test.tld:5", EndpointUtils.toAutoIpEndpoint("path/to/svc", "task", 5, mockSchedulerConfig));
-        assertEquals("task.pathtosvc-with-dots.what.a.fun.test.tld:5", EndpointUtils.toAutoIpEndpoint("path/to/svc.with.dots", "task", 5, mockSchedulerConfig));
+        assertEquals("task.svc.what.a.fun.test.tld:5",
+                EndpointUtils.toAutoIpEndpoint("svc", "task", 5, mockSchedulerConfig));
+        assertEquals("task.pathtosvc.what.a.fun.test.tld:5",
+                EndpointUtils.toAutoIpEndpoint("/path/to/svc", "task", 5, mockSchedulerConfig));
+        assertEquals("task.pathtosvc.what.a.fun.test.tld:5",
+                EndpointUtils.toAutoIpEndpoint("path/to/svc", "task", 5, mockSchedulerConfig));
+        assertEquals("task.pathtosvc-with-dots.what.a.fun.test.tld:5",
+                EndpointUtils.toAutoIpEndpoint("path/to/svc.with.dots", "task", 5, mockSchedulerConfig));
     }
 
     @Test
     public void testToVipEndpoint() {
-        assertEquals("vip.svc.l4lb.thisdcos.directory:5", EndpointUtils.toVipEndpoint("svc", new VipInfo("vip", 5)));
-        assertEquals("vip.pathtosvc.l4lb.thisdcos.directory:5", EndpointUtils.toVipEndpoint("/path/to/svc", new VipInfo("vip", 5)));
-        assertEquals("vip.pathtosvc.l4lb.thisdcos.directory:5", EndpointUtils.toVipEndpoint("path/to/svc", new VipInfo("vip", 5)));
-        assertEquals("vip.pathtosvc.with.dots.l4lb.thisdcos.directory:5", EndpointUtils.toVipEndpoint("path/to/svc.with.dots", new VipInfo("vip", 5)));
+        assertEquals("vip.svc.vip.tld:5",
+                EndpointUtils.toVipEndpoint("svc", mockSchedulerConfig, new VipInfo("vip", 5)));
+        assertEquals("vip.pathtosvc.vip.tld:5",
+                EndpointUtils.toVipEndpoint("/path/to/svc", mockSchedulerConfig, new VipInfo("vip", 5)));
+        assertEquals("vip.pathtosvc.vip.tld:5",
+                EndpointUtils.toVipEndpoint("path/to/svc", mockSchedulerConfig, new VipInfo("vip", 5)));
+        assertEquals("vip.pathtosvc.with.dots.vip.tld:5",
+                EndpointUtils.toVipEndpoint("path/to/svc.with.dots", mockSchedulerConfig, new VipInfo("vip", 5)));
 
-        assertEquals("vip.svc.l4lb.thisdcos.directory:5", EndpointUtils.toVipEndpoint("svc", new VipInfo("/vip", 5)));
-        assertEquals("vip.pathtosvc.l4lb.thisdcos.directory:5", EndpointUtils.toVipEndpoint("/path/to/svc", new VipInfo("/vip", 5)));
-        assertEquals("vip.pathtosvc.l4lb.thisdcos.directory:5", EndpointUtils.toVipEndpoint("path/to/svc", new VipInfo("/vip", 5)));
-        assertEquals("vip.pathtosvc.with.dots.l4lb.thisdcos.directory:5", EndpointUtils.toVipEndpoint("path/to/svc.with.dots", new VipInfo("/vip", 5)));
+        assertEquals("vip.svc.vip.tld:5",
```

Comment:
The intent of the test cases from here on are not really clear. I mean, basically we're checking that the VIP endpoint doesn't care about the leading slash in `VipInfo`. Would it not make sense to just break this out into a separate test?